### ISSUE TITLE
rollout-check jobs causes argo out-of-sync

### ISSUE THREAD

**Issue by huzlak** [solo-io/gloo#8818]
### Gloo Edge Product

Enterprise

### Gloo Edge Version

1.14.11

### Kubernetes Version

v1.24.0

### Describe the bug

Upgrade to 1.14.11 causes argoCD app to be constantly out-of-sync. 

What seems to be the problem:
- the gloo-esource-rollout-job is no longer a post-sync step
- this makes ArgoCD consider it as a resource to track and ensure it is in sync with what’s in Git
- the gloo-esource-rollout-job-check eventually delete the gloo-esource-rollout-job as a post-sync action
- this makes ArgoCD complain that gloo-resource-rollout-job does not exist while Git says it should
The same problem occurs with gloo-ee-resource-rollout-job

### Expected Behavior

Jobs should not cause out-of-sync app state

### Steps to reproduce the bug

Deploy gloo-ee helm chart with argo

### Additional Context

Setting below annotations on the job resource did not help:
```
argocd.argoproj.io/hook-delete-policy: HookSucceeded
argocd.argoproj.io/sync-wave: '10'
argocd.argoproj.io/compare-options: IgnoreExtraneous
```

**Comment by sam-heilbron** [comment]
I believe this is a duplicate of https://github.com/solo-io/gloo/issues/8809

In our slack post, we identify 1.14.11 as one of the affected versions: https://solo-io-corp.slack.com/archives/CEDCS8TAP/p1697744028710359

**Comment by huzlak** [comment]
I created this to make sure this specific problem is not present once fix for custom readiness probe is re-introduced. the problem reported in https://github.com/solo-io/gloo/issues/8809 is about pods crashing. here the out-of-sync is not caused by the crash itself.

**Comment by davidjumani** [comment]
Thanks for creating this @huzlak I'll ensure that deployments via argo do not go out of sync due to this

**Comment by davidjumani** [comment]
Fixed in [v1.16.0-beta24](https://github.com/solo-io/gloo/releases/tag/v1.16.0-beta24) [v1.15.16](https://github.com/solo-io/gloo/releases/tag/v1.15.16) [v1.14.27](https://github.com/solo-io/gloo/releases/tag/v1.14.27) [v1.13.32](https://github.com/solo-io/gloo/releases/tag/v1.13.32)
@huzlak Can you please verify this is fixed in [v1.16.0-beta24](https://github.com/solo-io/gloo/releases/tag/v1.16.0-beta24) ?

**Comment by davidjumani** [comment]
The issue is fixed in the versions mentioned above. @huzlak Feel free to reopen if you face further issues
